src/librbd/ImageCtx.cc

```diff
@@ -298,9 +298,8 @@ namespace librbd {
       snap_exists = true;
       data_ctx.snap_set_read(snap_id);
 
-      if (object_map != NULL) {
-        object_map->refresh(in_snap_id);
-      }
+      RWLock::WLocker l(object_map_lock);
```

Comment:
Minor: Just for consistency sake w/ the other calls to ObjectMap methods, would it be possible to move ```RWLock::WLocker l(object_map_lock)``` to within the method itself?  Looking at the other places it is invoked, the lock is taken right before invoking this method.